backend/api/Controllers/InspectionFindingsController.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Api.Controllers.Models;
+using Api.Database.Models;
+using Api.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Api.Controllers
+{
+    [ApiController]
+    [Route("inspection-findings")]
+    public class InspectionFindingsController : ControllerBase
+    {
+        private readonly IInspectionService _inspectionService;
+        private readonly ILogger<InspectionFindingsController> _logger;
+        public InspectionFindingsController(
+            ILogger<InspectionFindingsController> logger,
+            IInspectionService inspectionService
+        )
+        {
+            _logger = logger;
+            _inspectionService = inspectionService;
+
+        }
+
+        /// <summary>
+        /// Associate a new inspection finding with the inspection corresponding to isarStepId
+        /// </summary>
+        /// <remarks>
+        /// </remarks>
+        [HttpPost("add-findings")]
+        [Authorize(Roles = Role.Admin)]
+        [ProducesResponseType(typeof(InspectionFindings), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status409Conflict)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult<InspectionFindings>> AddFindings([FromBody] InspectionFindingsQuery inspectionFinding)
+        {
+            _logger.LogInformation("Updating inspection findings for inspection with isarStepId '{Id}'", inspectionFinding.IsarStepId);
+            try
+            {
+                var inspection = await _inspectionService.ReadByIsarStepId(inspectionFinding.IsarStepId);
+                if (inspection != null)
+                {
+                    inspection = await _inspectionService.AddFindings(inspectionFinding);
+                    return Ok(inspection?.InspectionFindings);
```

Comment:
```suggestion
                    return Ok(inspection.InspectionFindings);
```
No need for a question mark here I think, as we have checked that inspection is not null.

Reply:
Nope it complains and throws a warning otherwise.

Reply:
I will see again.

Reply:
Because you already have inspection on line 43, then you override it on line 46

Reply:
This is handled in the lattest ammends.